masonry/src/widgets/radio_button.rs

```diff
@@ -0,0 +1,437 @@
+        Self {
+            checked,
+            label: label.to_pod(),
+        }
+    }
+}
+
+// --- MARK: WIDGETMUT
+impl RadioButton {
+    /// Check or uncheck the box.
+    pub fn set_checked(this: &mut WidgetMut<'_, Self>, checked: bool) {
+        this.widget.checked = checked;
+        // Checked state impacts appearance and accessibility node
+        this.ctx.request_render();
+    }
+
+    /// Set the text.
+    ///
+    /// We enforce this to be an `ArcStr` to make the allocation explicit.
+    pub fn set_text(this: &mut WidgetMut<'_, Self>, new_text: ArcStr) {
+        Label::set_text(&mut Self::label_mut(this), new_text);
+    }
+
+    /// Get a mutable reference to the label.
+    pub fn label_mut<'t>(this: &'t mut WidgetMut<'_, Self>) -> WidgetMut<'t, Label> {
+        this.ctx.get_mut(&mut this.widget.label)
+    }
+}
+
+impl HasProperty<DisabledBackground> for RadioButton {}
+impl HasProperty<ActiveBackground> for RadioButton {}
+impl HasProperty<Background> for RadioButton {}
+impl HasProperty<HoveredBorderColor> for RadioButton {}
+impl HasProperty<BorderColor> for RadioButton {}
+impl HasProperty<BorderWidth> for RadioButton {}
+impl HasProperty<CornerRadius> for RadioButton {}
+impl HasProperty<Padding> for RadioButton {}
+impl HasProperty<DisabledCheckmarkColor> for RadioButton {}
+impl HasProperty<CheckmarkColor> for RadioButton {}
+
+/// The action type emitted by [`RadioButton`] when it is activated.
+///
+/// The field is the target toggle state (i.e. true is "this radio button would like to become checked").
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct RadioButtonToggled;
+
+// --- MARK: IMPL WIDGET
+impl Widget for RadioButton {
+    type Action = RadioButtonToggled;
+
+    fn on_pointer_event(
+        &mut self,
+        ctx: &mut EventCtx<'_>,
+        _props: &mut PropertiesMut<'_>,
+        event: &PointerEvent,
+    ) {
+        match event {
+            PointerEvent::Down { .. } => {
+                ctx.capture_pointer();
+                trace!("RadioButton {:?} pressed", ctx.widget_id());
+            }
+            PointerEvent::Up { .. } => {
+                if ctx.is_active() && ctx.is_hovered() {
+                    ctx.submit_action::<Self::Action>(RadioButtonToggled);
+                    trace!("RadioButton {:?} released", ctx.widget_id());
+                }
+            }
+            _ => (),
+        }
+    }
+
+    fn on_text_event(
+        &mut self,
+        ctx: &mut EventCtx<'_>,
+        _props: &mut PropertiesMut<'_>,
+        event: &TextEvent,
+    ) {
+        match event {
+            TextEvent::Keyboard(event) if event.state.is_up() => {
+                if matches!(&event.key, Key::Character(c) if c == " ") {
+                    ctx.submit_action::<Self::Action>(RadioButtonToggled);
+                }
+            }
+            _ => (),
+        }
+    }
+
+    fn accepts_focus(&self) -> bool {
+        // RadioButton can be tab-focused...
+        true
```

Comment:
Well... Really, only the currently checked radio button can be tab focused, see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/radiogroup_role#tab.

This can be a TODO, as I believe that `accepts_focus` can't be toggled dynamically.